aggregator/src/aggregator/collection_job_driver.rs

```diff
@@ -48,9 +48,15 @@ pub struct CollectionJobDriver<B> {
 
     // Configuration.
     batch_aggregation_shard_count: u64,
-    // The minimum duration to wait before retrying a collection job that has been stepped but was
-    // not ready yet because not all included reports had finished aggregation.
+    /// The minimum duration to wait before retrying a collection job that has been stepped but was
+    /// not ready yet because of pending aggregation.
     min_collection_job_retry_delay: Duration,
+    /// The maximum duration to wait before retrying a collection job that has been stepped but was
+    /// not ready yet because of pending aggregation.
+    max_collection_job_retry_delay: Duration,
+    /// The exponential factor used when computing the next retry delay for a collection job that
+    /// has been stepped but was not ready yet because of pending aggregation.
+    collection_job_retry_delay_exponential_factor: f64,
```

Comment:
nit: you could factor these three values into a `struct JobRetryStrategy` or something, and then `compute_reacquire_delay` becomes a method on that. It would then be incrementally easier to adopt this exponential backoff logic in the aggregation job driver.

Reply:
Sure, YAGNI is a risk but we'll plausibly use this again -- I factored into a simple `RetryStrategy` type.